### PR TITLE
HTTP test helpers refactoring.

### PR DESCRIPTION
**Pull request by frankban**
Mostly mechanical.
Introduce DoRequest params.
s/expectCode/expectStatus/ in tests.
Make AssertJSONCall also check the response content type.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/167/

File: internal/v4/archive_test.go [hunk 1]
```diff
@@ -673,12 +687,12 @@ func (s *ArchiveSuite) TestDeleteSpecificCharm(c *gc.C) {
...
-		Method:     "DELETE",
-		Username:   serverParams.AuthUsername,
-		Password:   serverParams.AuthPassword,
-		ExpectCode: http.StatusOK,
+		Handler:      s.srv,
+		URL:          storeURL("utopic/mysql-42/archive"),
+		Method:       "DELETE",
+		Username:     serverParams.AuthUsername,
+		Password:     serverParams.AuthPassword,
+		ExpectStatus: http.StatusOK,
```
Comment (rogpeppe) [inline internal/v4/archive_test.go:695]: Thanks for changing the name of this field too. It's been bugging me for a while :-)

**Comment by rogpeppe** [comment]
LGTM. Thanks a lot.

**Comment by urosj** [comment]
:+1: Nice!!!

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**